In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv("brfss_clean_2020_2024.csv")
df.shape

(1244709, 11)

In [2]:
df.head()

,_STATE,_LLCPWT,_BMI5,_AGEG5YR,_SEX,_EDUCAG,_INCOMG1,_RACE,BMI,obese,year
0,1.0,284.335672,1660.0,8.0,2.0,4.0,1.0,1.0,16.60,0,2020
1,1.0,616.146154,2658.0,10.0,1.0,2.0,3.0,1.0,26.58,0,2020
2,1.0,341.379354,2421.0,12.0,2.0,2.0,4.0,1.0,24.21,0,2020
3,1.0,333.450142,2657.0,10.0,2.0,2.0,3.0,2.0,26.57,0,2020
4,1.0,1627.016526,2371.0,5.0,2.0,4.0,4.0,1.0,23.71,0,2020


In [3]:
df.columns.tolist()

['_STATE',
 '_LLCPWT',
 '_BMI5',
 '_AGEG5YR',
 '_SEX',
 '_EDUCAG',
 '_INCOMG1',
 '_RACE',
 'BMI',
 'obese',
 'year']

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1244709 entries, 0 to 1244708
Data columns (total 11 columns):
 #   Column    Non-Null Count    Dtype  
---  ------    --------------    -----  
 0   _STATE    1244709 non-null  float64
 1   _LLCPWT   1244709 non-null  float64
 2   _BMI5     1244709 non-null  float64
 3   _AGEG5YR  1244709 non-null  float64
 4   _SEX      1244709 non-null  float64
 5   _EDUCAG   1244709 non-null  float64
 6   _INCOMG1  1244709 non-null  float64
 7   _RACE     1244665 non-null  float64
 8   BMI       1244709 non-null  float64
 9   obese     1244709 non-null  int64  
 10  year      1244709 non-null  int64  
dtypes: float64(9), int64(2)
memory usage: 104.5 MB


# BRFSS Cleaned Dataset Reference

This notebook summarizes the cleaned BRFSS 2020–2024 dataset used in the project.

It is intended as a reference for:
- variable definitions
- category mappings
- year-wise coverage
- obesity outcome construction
- demographic structure used for modeling

In [5]:
df["year"].value_counts().sort_index()

year
2020    300259
2021    239678
2022    235582
2023    227847
2024    241343
Name: count, dtype: int64

In [6]:
df.groupby("year")["obese"].mean()

year
2020    0.330072
2021    0.363617
2022    0.361836
2023    0.357679
2024    0.364651
Name: obese, dtype: float64

In [7]:
main_cols = [
    "_STATE",
    "_LLCPWT",
    "_BMI5",
    "_AGEG5YR",
    "_SEX",
    "_EDUCAG",
    "_INCOMG1",
    "_RACE",
    "BMI",
    "obese",
    "year"
]

df[main_cols].head()

,_STATE,_LLCPWT,_BMI5,_AGEG5YR,_SEX,_EDUCAG,_INCOMG1,_RACE,BMI,obese,year
0,1.0,284.335672,1660.0,8.0,2.0,4.0,1.0,1.0,16.60,0,2020
1,1.0,616.146154,2658.0,10.0,1.0,2.0,3.0,1.0,26.58,0,2020
2,1.0,341.379354,2421.0,12.0,2.0,2.0,4.0,1.0,24.21,0,2020
3,1.0,333.450142,2657.0,10.0,2.0,2.0,3.0,2.0,26.57,0,2020
4,1.0,1627.016526,2371.0,5.0,2.0,4.0,4.0,1.0,23.71,0,2020


In [8]:
age_map = {
    1: "18-24",
    2: "25-29",
    3: "30-34",
    4: "35-39",
    5: "40-44",
    6: "45-49",
    7: "50-54",
    8: "55-59",
    9: "60-64",
    10: "65-69",
    11: "70-74",
    12: "75-79",
    13: "80+"
}

sex_map = {
    1: "Male",
    2: "Female"
}

education_map = {
    1: "Did not graduate high school",
    2: "Graduated high school",
    3: "Attended college or technical school",
    4: "Graduated college or technical school"
}

income_map = {
    1: "<15k",
    2: "15k-25k",
    3: "25k-35k",
    4: "35k-50k",
    5: "50k+"
}

race_map = {
    1: "White",
    2: "Black",
    3: "Other / Multiracial",
    4: "Asian",
    5: "Hispanic"
}

In [9]:
df["age_group"] = df["_AGEG5YR"].map(age_map)
df["sex"] = df["_SEX"].map(sex_map)
df["education"] = df["_EDUCAG"].map(education_map)
df["income_group"] = df["_INCOMG1"].map(income_map)
df["race_group"] = df["_RACE"].map(race_map)

In [10]:
print("Age groups:")
print(sorted(df["age_group"].dropna().unique()))

print("\nSex groups:")
print(sorted(df["sex"].dropna().unique()))

print("\nEducation groups:")
print(sorted(df["education"].dropna().unique()))

print("\nIncome groups:")
print(sorted(df["income_group"].dropna().unique()))

print("\nRace groups:")
print(sorted(df["race_group"].dropna().unique()))

Age groups:
['18-24', '25-29', '30-34', '35-39', '40-44', '45-49', '50-54', '55-59', '60-64', '65-69', '70-74', '75-79', '80+']

Sex groups:
['Female', 'Male']

Education groups:
['Attended college or technical school', 'Did not graduate high school', 'Graduated college or technical school', 'Graduated high school']

Income groups:
['15k-25k', '25k-35k', '35k-50k', '50k+', '<15k']

Race groups:
['Asian', 'Black', 'Hispanic', 'Other / Multiracial', 'White']


In [11]:
df["age_group"].value_counts().sort_index()

age_group
18-24     74009
25-29     68214
30-34     73834
35-39     76434
40-44     76861
45-49     73931
50-54     85900
55-59    100810
60-64    127960
65-69    143127
70-74    134207
75-79     99392
80+      110030
Name: count, dtype: int64

In [12]:
df["sex"].value_counts()

sex
Female    656069
Male      588640
Name: count, dtype: int64

In [13]:
df["education"].value_counts()

education
Graduated college or technical school    438765
Attended college or technical school     371334
Graduated high school                    354072
Did not graduate high school              80538
Name: count, dtype: int64

In [14]:
df["income_group"].value_counts().sort_index()

income_group
15k-25k    169659
25k-35k    183386
35k-50k    220866
50k+       569900
<15k       100898
Name: count, dtype: int64

In [15]:
df["race_group"].value_counts()

race_group
White                  899575
Black                  103330
Hispanic                30393
Other / Multiracial     30260
Asian                   28344
Name: count, dtype: int64

### Obesity Prevalence by Group

In [16]:
df.groupby("sex")["obese"].mean().sort_values(ascending=False)

sex
Female    0.368507
Male      0.338468
Name: obese, dtype: float64

In [17]:
df.groupby("age_group")["obese"].mean().sort_index()

age_group
18-24    0.214758
25-29    0.311769
30-34    0.359550
35-39    0.391632
40-44    0.412992
45-49    0.434324
50-54    0.437567
55-59    0.416248
60-64    0.391591
65-69    0.372257
70-74    0.347478
75-79    0.309140
80+      0.210942
Name: obese, dtype: float64

In [18]:
df.groupby("education")["obese"].mean().sort_values(ascending=False)

education
Did not graduate high school             0.387991
Attended college or technical school     0.380722
Graduated high school                    0.378731
Graduated college or technical school    0.306043
Name: obese, dtype: float64

In [19]:
df.groupby("income_group")["obese"].mean().sort_index()

income_group
15k-25k    0.378182
25k-35k    0.364733
35k-50k    0.357719
50k+       0.336257
<15k       0.389621
Name: obese, dtype: float64

In [20]:
df.groupby("race_group")["obese"].mean().sort_values(ascending=False)

race_group
Black                  0.448795
Hispanic               0.393084
Other / Multiracial    0.378354
White                  0.344289
Asian                  0.196161
Name: obese, dtype: float64

In [21]:
group_cols = ["age_group", "sex", "education", "income_group", "race_group"]

brfss_group_summary = (
    df.groupby(group_cols)
      .agg(
          obesity_rate=("obese", "mean"),
          n=("obese", "size")
      )
      .reset_index()
      .sort_values("n", ascending=False)
)

brfss_group_summary.head(20)

,age_group,sex,education,income_group,race_group,obesity_rate,n
2169,70-74,Male,Graduated college or technical school,50k+,White,0.298610,14966
1869,65-69,Female,Graduated college or technical school,50k+,White,0.285947,14737
2069,70-74,Female,Graduated college or technical school,50k+,White,0.262396,14360
1969,65-69,Male,Graduated college or technical school,50k+,White,0.308346,13180
1669,60-64,Female,Graduated college or technical school,50k+,White,0.306990,12590
2369,75-79,Male,Graduated college or technical school,50k+,White,0.259671,11426
1469,55-59,Female,Graduated college or technical school,50k+,White,0.325129,10482
2569,80+,Male,Graduated college or technical school,50k+,White,0.173921,10453
1269,50-54,Female,Graduated college or technical school,50k+,White,0.347884,9805
1769,60-64,Male,Graduated college or technical school,50k+,White,0.335160,9670


## Why this matters

BRFSS provides obesity outcomes by demographic group.

The ACS team needs to build county-level demographic population distributions using the same bins:

- age_group
- sex
- education
- income_group
- race_group

These two datasets will later be connected through post-stratification.

In [22]:
summary_table = pd.DataFrame({
    "Metric": [
        "Total rows",
        "Total years",
        "First year",
        "Last year",
        "Overall obesity prevalence"
    ],
    "Value": [
        len(df),
        df["year"].nunique(),
        df["year"].min(),
        df["year"].max(),
        round(df["obese"].mean(), 4)
    ]
})

summary_table

,Metric,Value
0,Total rows,1.244709e+06
1,Total years,5.000000e+00
2,First year,2.020000e+03
3,Last year,2.024000e+03
4,Overall obesity prevalence,3.543000e-01


## Final Summary of BRFSS Dataset (2020–2024)

The cleaned BRFSS dataset consists of approximately **1.24 million observations** across five years (2020–2024), with a consistent set of demographic and health-related variables used for modeling obesity prevalence. The dataset includes key features such as age group, sex, education level, income group, and race, along with a derived BMI variable and a binary obesity indicator.

Across all years, the dataset shows stable sample sizes (~230k–300k per year) and an overall obesity prevalence of approximately **35.4%**, with a noticeable increase from 2020 (~33%) to later years (~36%). Demographic distributions are well-balanced, with slightly more female respondents (~53%) and a majority of individuals falling into higher income and education categories.

Clear patterns emerge from the exploratory analysis:
- Obesity prevalence increases with age, peaking around 45–54 and declining in older age groups.
- Females show slightly higher obesity prevalence (~36.8%) compared to males (~33.8%).
- Lower income and lower education groups have higher obesity rates, indicating strong socioeconomic gradients.
- Racial disparities are evident, with the highest prevalence observed among Black individuals (~44.9%) and the lowest among Asians (~19.6%).

The dataset was further structured into demographic groups defined by combinations of age, sex, education, income, and race. These group-level summaries (including obesity rates and sample sizes) form the foundation for downstream modeling and will be used in conjunction with ACS demographic distributions for county-level estimation.